File: backend/mela/src/main/java/com/hcmus/mela/exercise/model/Option.java
package com.hcmus.mela.exercise.model;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document
public class Option {
    @Field(name = "ordinal_number")
    private Integer ordinalNumber;

    @Field(name = "content")
    private String content;

    @Field(name = "is_correct")
    private Boolean isCorrect;
}

File: frontend/mela/lib/domain/entity/question/option.dart
class Option {
  final int ordinalNumber;
  final String content;
  final bool isCorrect;

  Option({
    required this.ordinalNumber,
    required this.content,
    required this.isCorrect,
  });

  factory Option.fromJson(Map<String, dynamic> json) {
    return Option(
      ordinalNumber: json['ordinalNumber'],
      content: json['content'],
      isCorrect: json['isCorrect'],
    );
  }
}